pkg/client/cache/eventqueue.go

```diff
@@ -282,6 +289,13 @@ func (eq *EventQueue) Pop() (watch.EventType, interface{}, error) {
 		eventType := eq.events[key]
 		delete(eq.events, key)
 
+		// Track the last replace key immediately after the store
+		// state has been changed to prevent subsequent errors from
+		// leaving a stale key.
+		if eq.lastReplaceKey != "" && eq.lastReplaceKey == key {
```

Comment:
not important, making sure I'm not missing anything, but the `eq.lastReplaceKey != ""` isn't really doing anything, right?